File: fibonacci.rs
//usr/bin/env rustc $0 -o a.out && ./a.out && rm ./a.out ; exit
use std::time::Instant;

fn fibonacci(n: u32) -> u32 {
  match n {
    0 | 1 | 2 => 1,
    _ => fibonacci(n - 1) + fibonacci(n - 2),
  }
}

fn main() {
  let start = Instant::now();
  fibonacci(40);
  let elapsed = start.elapsed().as_millis() as f64;
  println!("rust..... {:.6} seconds", 0.001 * elapsed);
}

File: fibonacci.exs
#!/usr/bin/env elixir

defmodule Main do 
  def fibonacci(1) do 1 end
  def fibonacci(2) do 1 end
  def fibonacci(n) do fibonacci(n-1) + fibonacci(n-2) end
end

start = :os.system_time(:millisecond)
Main.fibonacci(40)
secs = (:os.system_time(:millisecond) - start) * 0.001
IO.puts "elixir... #{Float.ceil(secs, 6)} seconds"
